contracts/FileStorage.sol

```diff
@@ -52,10 +52,15 @@ contract FileStorage is AccessControlEnumerableUpgradeable {
         bool[] isChunkUploaded;
     }
 
+    struct DetailedInfo {
+        bool isImmutable;
+    }
+
     struct Directory {
         ContentInfo[] contents;
         mapping(string => uint) contentIndexes;
         mapping(string => Directory) directories;
+        mapping(string => DetailedInfo) contentDetails;
```

Comment:
It would be cheaper to use not dynamic type as a key. For example `bytes32` and each time transform `string` to `bytes32` by `keccak256`